cl/phase1/network/services/aggregate_and_proof_service.go

```diff
@@ -147,134 +280,166 @@ func (a *aggregateAndProofServiceImpl) ProcessMessage(
 		return errors.New("invalid aggregate and proof")
 	}
 	attestingIndicies, err := headState.GetAttestingIndicies(
-		aggregateAndProof.Message.Aggregate.AttestantionData(),
-		aggregateAndProof.Message.Aggregate.AggregationBits(),
+		aggregateAndProof.SignedAggregateAndProof.Message.Aggregate.AttestantionData(),
+		aggregateAndProof.SignedAggregateAndProof.Message.Aggregate.AggregationBits(),
 		true,
 	)
 	if err != nil {
 		return err
 	}
-	if err := verifySignaturesOnAggregate(headState, aggregateAndProof); err != nil {
-		return err
-	} // Add to aggregation pool
-	a.opPool.AttestationsPool.Insert(
-		aggregateAndProof.Message.Aggregate.Signature(),
-		aggregateAndProof.Message.Aggregate,
-	)
-	a.forkchoiceStore.ProcessAttestingIndicies(
-		aggregateAndProof.Message.Aggregate,
-		attestingIndicies,
-	)
-	return nil
-}
+	if len(attestingIndicies) == 0 {
+		return errors.New("no attesting indicies")
+	}
 
-func verifySignaturesOnAggregate(
-	s *state.CachingBeaconState,
-	aggregateAndProof *cltypes.SignedAggregateAndProof,
-) error {
-	aggregationBits := aggregateAndProof.Message.Aggregate.AggregationBits()
-	// [REJECT] The aggregate attestation has participants -- that is, len(get_attesting_indices(state, aggregate)) >= 1.
-	attestingIndicies, err := s.GetAttestingIndicies(
-		aggregateAndProof.Message.Aggregate.AttestantionData(),
-		aggregationBits,
-		true,
-	)
+	// aggregate signatures for later verification
+	aggregateVerificationData, err := GetSignaturesOnAggregate(headState, aggregateAndProof.SignedAggregateAndProof, attestingIndicies)
 	if err != nil {
 		return err
 	}
-	if len(attestingIndicies) == 0 {
-		return errors.New("no attesting indicies")
+
+	// further processing will be done after async signature verification
+	aggregateVerificationData.F = func() {
+		a.opPool.AttestationsPool.Insert(
```

Comment:
Can we just do 
```
(*aggregateAndProofServiceImpl).opPool.AttestationsPool.Insert(...)
(*aggregateAndProofServiceImpl).forkchoiceStore.ProcessAttestingIndicies(...)
```
after verification done? I mean, seems like we don't need to setup callback function.